application/src/main/java/org/togetherjava/tjbot/features/reminder/RemindRoutine.java

```diff
@@ -147,34 +134,29 @@ static ReminderRoute toPrivate(PrivateChannel channel) {
         }
     }
 
-    private void reminderFailureAction(int id, long channelId, long authorId,
-                                       CharSequence content, long guildId, int reminderFailureCounter) {
-        if (reminderFailureCounter <= 3) {
-            Instant current = Instant.now();
-            database.write(context -> context.newRecord(PENDING_REMINDERS)
-                    .setId(id)
-                    .setCreatedAt(Instant.now())
-                    .setGuildId(guildId)
-                    .setChannelId(channelId)
-                    .setAuthorId(authorId)
-                    .setRemindAt(current.plusSeconds(60))
-                    .setContent(content.toString())
-                    .setReminderFailureCounter(reminderFailureCounter + 1)
-                    .insert());
-        } else {
+    private void attemptRetryReminder(Reminder reminder) {
+        if (!(reminder.failureAttempts() <= MAX_FAILURE_RETRY)) {
             logger.warn(
                     """
-                            Failed to send a reminder with (authorID '{}'), skipping it. This can be due to a network issue, \
+                            Failed to send a reminder with (authorID '{}') and (content '{}'), skipping it. This can be due to a network issue, \
                             but also happen if the bot disconnected from the target guild and the \
-                            user has disabled DMs or has been deleted."""+ "contents: "+content,
-                    authorId);
-        }
-    }
+                            user has disabled DMs or has been deleted.""",
+                    reminder.authorId(),reminder.content());
```

Comment:
imo the log should also log the failure exception. otherwise we cant investigate why it failed

Reply:
do i need to create a custom exception?

Reply:
no. just carry the `failure` through the method param and then attach it to the log as last parameter `logger.warn(..., failure)`. the logger will auto-attach the stacktrace if u put it as last argument

Reply:
i should prolly use `failure.getMessage()`? to get proper message from failure throwable

Reply:
(u did it wrong, see comment in newest CR)